context/src/test/groovy/io/micronaut/scheduling/ScheduledFixedRateSpec.groovy

```diff
@@ -79,8 +80,11 @@ class ScheduledFixedRateSpec extends Specification {
         MyOtherTask myTask = beanContext.getBean(MyOtherTask)
 
         then:
+        // Cron tasks run once every (30/31/32) seconds, starting with the application startup.
+        // A timeout of 20 seconds ensures that each task triggers once.
         conditions.eventually {
-            myTask.cronEvents.get() == 2
```

Comment:
This test has a bug. The timeout is 20 seconds, but the cron tasks run once every 3 seconds and once every 4 seconds, respectively. It is possible for the `cronEvents` counter to jump from 1 to 3 between the polling conditions.

I refactored it to ensure that each of these cron tasks trigger exactly once within the time window.

Reply:
Side note - if one of the two original scheduled cron tasks broke, this test wouldn't catch that. Instead the one working test would increment the counter once every 3 or 4 seconds, causing this original condition to resolve as true.